byron/semantics/executable-spec/src/Control/State/Transition/Generator.hs

```diff
@@ -140,9 +146,9 @@ allValid
     }
 
 generateSignalWithFailureProportions
-  :: [(Int, PredicateFailure s)]
+  :: [(Int, Constr)]
   -- ^ Failure proportions. See 'failures' in 'TraceProfile'.
-  -> (Maybe (PredicateFailure s) -> Environment s -> State s -> Gen (Signal s))
+  -> (Maybe Constr -> Environment s -> State s -> Gen (Signal s))
```

Comment:
See my above comment. I think we can actually change these back to `PredicateFailure s` and use `eqPF` to compare "PredicateFailure-constructor-equality".